plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/cnf/ServerModule.java

```diff
@@ -185,5 +191,46 @@ public void restart(){
 		Class<?>[] classes = {TreeItem.class, ServersView2.class};
 		return classes;
 	}
+	
+	/**
+	 * Get direct child modules
+	 * 
+	 * @return List of modules, which are direct module children
+	 */
+	public List<ServerModule> getModules() {
+		List<ServerModule> childModules = new ArrayList<ServerModule>();
+		List<TreeItem> childModuleTreeItems = this.treeItem.getItems();
+		for (TreeItem item : childModuleTreeItems) {
+			childModules.add(new ServerModule(item, view));
+		}
+		return childModules;
+	}
+	
+	/**
+	 * Returns module's name
+	 * 
+	 * @return the module's name
+	 */
+	public String getName() {
+		return getTreeItem().getText().split(" ")[0];
+	}
+	
+	/**
+	 * Returns module's TreeItem object
+	 * 
+	 * @return the module's TreeItem object
+	 */
+	public TreeItem getTreeItem() {
+		return this.treeItem;
+	}
+	
+	/**
+	 * Check if server module is synchronized
+	 * 
+	 * @return true if server module is synchronized, false otherwise
+	 */
+	public boolean isSynchronized() {
```

Comment:
I do not see this used anywhere or tests for this. You can use this enum for checking publish state: https://github.com/eclipse/reddeer/blob/master/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/cnf/ServersViewEnums.java#L86

Reply:
I understand, removed.